### PR TITLE
Update SDK

### PR DESCRIPTION
**Pull request by surayya-MS**
https://github.com/dotnet/aspnetcore/pull/47537

### CODE REVIEW

**Comment by surayya-MS** [comment]
In the closed PR I changed only global.js file. There were a lot of exceptions saying "Use 'ArgumentOutOfRangeException.ThrowIfEqual' instead of explicitly throwing a new exception instance":
.https://github.com/dotnet/aspnetcore/pull/47537/files 

I updated SDK and used ArgumentOutOfRangeException.ThrowIfEqual in this PR. Now checks are failing saying "'ArgumentOutOfRangeException' does not contain a definition for 'ThrowIfEqual'" 

@BrennanConroy , could you please help me?

**Comment by JamesNK** [comment]
> Well `ThrowIfEqual` doesn't even exist yet... [dotnet/runtime#83853](https://github.com/dotnet/runtime/pull/83853)
> 
> We should ignore the rule in our .editorconfig file and file an issue to fix it when the api exists.

It looks like just one warning. IMO, suppress the warning in *.cs file with a pragma. That isolate suppression to just places that have a bad warning. Add a follow-up issue to re-enable once `ThrowIfEqual` is available. Link to issue from cs file.

**Comment by BrennanConroy** [comment]
Oh good, now Helix is using the broken version of runtime D:
https://github.com/dotnet/runtime/pull/84009

File: src/Hosting/Server.IntegrationTesting/src/Deployers/RemoteWindowsDeployer/RemoteWindowsDeployer.cs
Comment (BrennanConroy) [inline src/Hosting/Server.IntegrationTesting/src/Deployers/RemoteWindowsDeployer/RemoteWindowsDeployer.cs:127]: Why not fix it instead?
I assume it's supposed to be `LogWarning(0, ex, "...")`

**Comment by surayya-MS** [comment]
After updating SDK to a newer version 8.0.100-preview.4.23204.1 the CA1512 suppression is no longer needed. It was the one causing warning in `src/SignalR/common/Http.Connections/src/HttpConnectionDispatcherOptions.cs`.

Now there is a new warning CA2017 "Number of parameters supplied in the logging message template do
not match the number of named placeholders." in `\src\Hosting\Server.IntegrationTesting\src\Deployers\RemoteWindowsDep
loyer\RemoteWindowsDeployer.cs`. Suppressing that.

File: src/Hosting/Server.IntegrationTesting/src/Deployers/RemoteWindowsDeployer/RemoteWindowsDeployer.cs
Comment (surayya-MS) [reply src/Hosting/Server.IntegrationTesting/src/Deployers/RemoteWindowsDeployer/RemoteWindowsDeployer.cs:127]: Thanks! I will change it to `LogWarning(0, ex, "...")` then.

Comment (surayya-MS) [reply src/Hosting/Server.IntegrationTesting/src/Deployers/RemoteWindowsDeployer/RemoteWindowsDeployer.cs:127]: Could you please file an issue for fixing `$""`? You would describe it better than me since you have more context.

**Review by JamesNK — APPROVED** [review]

File: src/SignalR/common/Http.Connections/src/HttpConnectionDispatcherOptions.cs
Comment (BrennanConroy) [inline src/SignalR/common/Http.Connections/src/HttpConnectionDispatcherOptions.cs:111]: Shouldn't this be fixable now since the APIs exist?

Comment (MackinnonBuck) [reply src/SignalR/common/Http.Connections/src/HttpConnectionDispatcherOptions.cs:111]: I tried this but it looks like they still haven't been added?

Comment (wtgodbe) [reply src/SignalR/common/Http.Connections/src/HttpConnectionDispatcherOptions.cs:111]: Something's up with maestro, we haven't had a dependency PR opened here in 4 days. I'll post in FR (CC @dotnet/dnceng)

Comment (wtgodbe) [reply src/SignalR/common/Http.Connections/src/HttpConnectionDispatcherOptions.cs:111]: They're investigating (it's a timeout in darc), in the meantime I manually triggered the subscriptions.

**Comment by mitchdenny** [comment]
Just rotating on to build ops - so are we in a holding pattern for this?

**Comment by javiercn** [comment]
@wtgodbe an update was pushed to the SDK to address this. It was an issue with a recent change.

We should be able to update the SDK version here, and that should get rid of that.

**Comment by wtgodbe** [comment]
https://github.com/dotnet/installer/pull/16148 will also unblock this - so whatever gets in first between that & https://github.com/dotnet/installer/pull/16033

**Comment by MichaelSimons** [comment]
> [dotnet/installer#16148](https://github.com/dotnet/installer/pull/16148) will also unblock this - so whatever gets in first between that & [dotnet/installer#16033](https://github.com/dotnet/installer/pull/16033)

FYI, https://github.com/dotnet/installer/pull/16148 was just merged.

**Comment by wtgodbe** [comment]
Nice! Once https://dev.azure.com/dnceng/internal/_build/results?buildId=2161581&view=results finishes we can use that SDK to unblock this PR. I'll keep an eye on it.

**Comment by wtgodbe** [comment]
Getting new errors like this:

> ##[error]/mnt/vss/_work/1/s/src/Components/benchmarkapps/Wasm.Performance/TestApp/Microsoft.NET.Sdk.Razor.SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.RazorSourceGenerator/Shared_PlainTable_TableComponent_razor.g.cs(106,146): error CS0400: (NETCORE_ENGINEERING_TELEMETRY=Build) The type or namespace name 'Pages' could not be found in the global namespace (are you missing an assembly reference?)
/mnt/vss/_work/1/s/src/Components/benchmarkapps/Wasm.Performance/TestApp/Shared/PlainTable/TableComponent.razor(13,31): error CS1503: Argument 1: cannot convert from 'Wasm.Performance.TestApp.Pages.GridRendering.WeatherForecast[]' to 'Pages.GridRendering.WeatherForecast[]' [/mnt/vss/_work/1/s/src/Components/benchmarkapps/Wasm.Performance/TestApp/Wasm.Performance.TestApp.csproj]
##[error]src/Components/benchmarkapps/Wasm.Performance/TestApp/Shared/PlainTable/TableComponent.razor(13,31): error CS1503: (NETCORE_ENGINEERING_TELEMETRY=Build) Argument 1: cannot convert from 'Wasm.Performance.TestApp.Pages.GridRendering.WeatherForecast[]' to 'Pages.GridRendering.WeatherForecast[]'
/mnt/vss/_work/1/s/src/Components/benchmarkapps/Wasm.Performance/TestApp/Microsoft.NET.Sdk.Razor.SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.RazorSourceGenerator/Shared_PlainTable_RowCollection_razor.g.cs(82,146): error CS0400: The type or namespace name 'Pages' could not be found in the global namespace (are you missing an assembly reference?) [/mnt/vss/_work/1/s/src/Components/benchmarkapps/Wasm.Performance/TestApp/Wasm.Performance.TestApp.csproj]
##[error]/mnt/vss/_work/1/s/src/Components/benchmarkapps/Wasm.Performance/TestApp/Microsoft.NET.Sdk.Razor.SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.RazorSourceGenerator/Shared_PlainTable_RowCollection_razor.g.cs(82,146): error CS0400: (NETCORE_ENGINEERING_TELEMETRY=Build) The type or namespace name 'Pages' could not be found in the global namespace (are you missing an assembly reference?)
/mnt/vss/_work/1/s/src/Components/benchmarkapps/Wasm.Performance/TestApp/Shared/PlainTable/RowCollection.razor(5,17): error CS1503: Argument 1: cannot convert from 'Wasm.Performance.TestApp.Pages.GridRendering.WeatherForecast' to 'Pages.GridRendering.WeatherForecast' [/mnt/vss/_work/1/s/src/Components/benchmarkapps/Wasm.Performance/TestApp/Wasm.Performance.TestApp.csproj]
##[error]src/Components/benchmarkapps/Wasm.Performance/TestApp/Shared/PlainTable/RowCollection.razor(5,17): error CS1503: (NETCORE_ENGINEERING_TELEMETRY=Build) Argument 1: cannot convert from 'Wasm.Performance.TestApp.Pages.GridRendering.WeatherForecast' to 'Pages.GridRendering.WeatherForecast'
/mnt/vss/_work/1/s/src/Components/benchmarkapps/Wasm.Performance/TestApp/Microsoft.NET.Sdk.Razor.SourceGenerators/Microsoft.NET.Sdk.Razor.SourceGenerators.RazorSourceGenerator/Shared_ComplexTable_TableComponent_razor.g.cs(115,152): error CS0400: The type or namespace name 'Pages' could not be found in the global namespace (are you missing an assembly reference?) [/mnt/vss/_work/1/s/src/Components/benchmarkapps/Wasm.Performance/TestApp/Wasm.Performance.TestApp.csproj]

@MackinnonBuck could you take a look?
@JamesNK this sort of looks like a code generation ordering issue that I think you helped fix a while back, does it ring a bell?

**Comment by JamesNK** [comment]
> @JamesNK this sort of looks like a code generation ordering issue that I think you helped fix a while back, does it ring a bell?

No 🤷‍♂️

**Comment by MackinnonBuck** [comment]
Looks like something broke with using aliases. In `TableComponent.razor`, for example:

```razor
@using WeatherForecast = Pages.GridRendering.WeatherForecast
```

In this case, the `Pages` namespace is relative to `Wasm.Performance.TestApp`. But then the Razor compiler emits:

```csharp
global::Microsoft.AspNetCore.Components.CompilerServices.RuntimeHelpers.TypeCheck<global::Pages.GridRendering.WeatherForecast[]>(...)
```

(Note `global::Pages`)

@chsienki Do you know what might have caused this change?

**Comment by chsienki** [comment]
@MackinnonBuck Hmm, I'm not sure on the exact PR, but we've definitely done a bunch of work amending `global::` to types, so I'm not totally surprised we're now incorrectly doing that. 

@jjonescz has been doing most of that work so hopefully he'll know. I've opened https://github.com/dotnet/razor/issues/8610 to track it.

**Comment by mitchdenny** [comment]
I've bumped the SDK version again. I noticed that on another branch I had some of the issues weren't present (I tried a newer SDK to resolve another issue). So its possible that this newer SDK might get us further.

**Comment by mitchdenny** [comment]
OK, that solved the user-jwts ListCommand issue, but we still have the Razor issue. It doesn't look like its been linked directly here yet, but this is the PR we are waiting on from Razor:

https://github.com/dotnet/razor/pull/8614

**Comment by wtgodbe** [comment]
The linked razor change will flow into dotnet/installer with https://github.com/dotnet/installer/pull/16240, so that's the next potential unblocker

**Comment by eerhardt** [comment]
The razor change is already in the recent SDK version I updated to. However the current failures are new:

```
CSC(0,0): error EnableGenerateDocumentationFile: (NETCORE_ENGINEERING_TELEMETRY=Build) Set MSBuild property 'GenerateDocumentationFile' to 'true' in project file to enable IDE0005 (Remove unnecessary usings/imports) on build (https://github.com/dotnet/roslyn/issues/41640)
```

**Comment by wtgodbe** [comment]
Ah, you're right. @sharwell @Youssef1313 any idea why the error Eric pasted above is starting to show up with the recent SDK update?

**Comment by Youssef1313** [comment]
@wtgodbe The PR https://github.com/dotnet/roslyn/pull/58835 was just merged yesterday. The newer .NET SDK seems to include it. The PR has details on this warning, but you can disable it if you don't care about IDE0005.

**Comment by Youssef1313** [comment]
Hmmm, I think this should be IDE-only diagnostic and not something that shows up in command-line builds since the diagnostic has `EnforceOnBuild.Never`. cc @mavasani

**Comment by eerhardt** [comment]
Should we give a better error / help link / etc? I'm not sure how a normal developer would ever understand what is happening here.

**Comment by sharwell** [comment]
IDE0005 is explicitly set to warning here, but the configuration is silently failing because the project is misconfigured.
https://github.com/dotnet/roslyn/issues/41640

Setting the value to `suggestion` will resolve the problem.

> Hmmm, I think this should be IDE-only diagnostic and not something that shows up in command-line builds since the diagnostic has `EnforceOnBuild.Never`. cc @mavasani

This is not possible because the IDE forces the use of a different documentation parse value than the build-time compiler does. The IDE-only analyzer will read that the diagnostic is working correctly, when in reality it is silently failing.

**Comment by mavasani** [comment]
@sharwell we can convert this new warning to only fire in the IDE and not at build time. That was the original intent and implementation in https://github.com/dotnet/roslyn/pull/58835. I suspect lot of developers will run into this as no one would know to enable doc comments for IDE0005 build enforcement and a build break does feel a bit of heavy hammer for such a misconfiguration

**Comment by Youssef1313** [comment]
> That was the original intent and implementation in [dotnet/roslyn#58835](https://github.com/dotnet/roslyn/pull/58835)

That is what I was referring to in https://github.com/dotnet/aspnetcore/pull/47540#issuecomment-1523817454, the implementation **already** uses `EnforceOnBuild.Never`, and I can't see why it's not working as intended.

**Comment by mitchdenny** [comment]
The following test failed: BlazorWasmHostedTemplate_IndividualAuth_NoHttps_Works_WithOutLocalDB 

Looking at the failure history it looks like it fails once a day or so on main so unrelated to this SDK update - we should consider disabling it. For now I've triggered that job again to see if we can get a pass.

/cc @dotnet/aspnet-blazor-eng

**Comment by mitchdenny** [comment]
PR to quarantine Blazor test: https://github.com/dotnet/aspnetcore/pull/47920

**Comment by mitchdenny** [comment]
Latest failure is some Blazor tests failing. Looks to be something to do with file locking.

**Comment by eerhardt** [comment]
@javiercn @lewing - It looks like the BlazorWASM template tests are failing with the latest SDK:

```
2023-04-27T14:13:49.7961892Z [xUnit.net 00:02:17.66]       /datadisks/disk1/work/B95509FE/p/dotnet-cli/sdk/8.0.100-preview.5.23226.2/Sdks/Microsoft.NET.Sdk.BlazorWebAssembly/targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets(160,5): error MSB4018: The "GenerateBlazorBootExtensionJson" task failed unexpectedly. [/datadisks/disk1/work/B95509FE/w/998E08AE/e/Templates/BaseFolder/AspNet.hhjg2t3oatk0/Client/AspNet.hhjg2t3oatk0.Client.csproj]
2023-04-27T14:13:49.7962111Z [xUnit.net 00:02:17.66]       /datadisks/disk1/work/B95509FE/p/dotnet-cli/sdk/8.0.100-preview.5.23226.2/Sdks/Microsoft.NET.Sdk.BlazorWebAssembly/targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets(160,5): error MSB4018: System.IO.IOException: The process cannot access the file '/datadisks/disk1/work/B95509FE/w/998E08AE/e/Templates/BaseFolder/AspNet.hhjg2t3oatk0/Client/obj/Release/net8.0/blazor.publish.boot-extension.json' because it is being used by another process. [/datadisks/disk1/work/B95509FE/w/998E08AE/e/Templates/BaseFolder/AspNet.hhjg2t3oatk0/Client/AspNet.hhjg2t3oatk0.Client.csproj]
2023-04-27T14:13:49.8009246Z [xUnit.net 00:02:17.66]       /datadisks/disk1/work/B95509FE/p/dotnet-cli/sdk/8.0.100-preview.5.23226.2/Sdks/Microsoft.NET.Sdk.BlazorWebAssembly/targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets(160,5): error MSB4018:    at Microsoft.Win32.SafeHandles.SafeFileHandle.Init(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Int64& fileLength, UnixFileMode& filePermissions) [/datadisks/disk1/work/B95509FE/w/998E08AE/e/Templates/BaseFolder/AspNet.hhjg2t3oatk0/Client/AspNet.hhjg2t3oatk0.Client.csproj]
2023-04-27T14:13:49.8760559Z [xUnit.net 00:02:17.66]       /datadisks/disk1/work/B95509FE/p/dotnet-cli/sdk/8.0.100-preview.5.23226.2/Sdks/Microsoft.NET.Sdk.BlazorWebAssembly/targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets(160,5): error MSB4018:    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, UnixFileMode openPermissions, Int64& fileLength, UnixFileMode& filePermissions, Func`4 createOpenException) [/datadisks/disk1/work/B95509FE/w/998E08AE/e/Templates/BaseFolder/AspNet.hhjg2t3oatk0/Client/AspNet.hhjg2t3oatk0.Client.csproj]
2023-04-27T14:13:49.8760938Z [xUnit.net 00:02:17.66]       /datadisks/disk1/work/B95509FE/p/dotnet-cli/sdk/8.0.100-preview.5.23226.2/Sdks/Microsoft.NET.Sdk.BlazorWebAssembly/targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets(160,5): error MSB4018:    at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode) [/datadisks/disk1/work/B95509FE/w/998E08AE/e/Templates/BaseFolder/AspNet.hhjg2t3oatk0/Client/AspNet.hhjg2t3oatk0.Client.csproj]
2023-04-27T14:13:49.8761071Z [xUnit.net 00:02:17.66]       /datadisks/disk1/work/B95509FE/p/dotnet-cli/sdk/8.0.100-preview.5.23226.2/Sdks/Microsoft.NET.Sdk.BlazorWebAssembly/targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets(160,5): error MSB4018:    at System.IO.File.Create(String path) [/datadisks/disk1/work/B95509FE/w/998E08AE/e/Templates/BaseFolder/AspNet.hhjg2t3oatk0/Client/AspNet.hhjg2t3oatk0.Client.csproj]
2023-04-27T14:13:49.8761121Z [xUnit.net 00:02:17.66]       /datadisks/disk1/work/B95509FE/p/dotnet-cli/sdk/8.0.100-preview.5.23226.2/Sdks/Microsoft.NET.Sdk.BlazorWebAssembly/targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets(160,5): error MSB4018:    at Microsoft.NET.Sdk.BlazorWebAssembly.GenerateBlazorBootExtensionJson.Execute() [/datadisks/disk1/work/B95509FE/w/998E08AE/e/Templates/BaseFolder/AspNet.hhjg2t3oatk0/Client/AspNet.hhjg2t3oatk0.Client.csproj]
2023-04-27T14:13:49.8761711Z [xUnit.net 00:02:17.66]       /datadisks/disk1/work/B95509FE/p/dotnet-cli/sdk/8.0.100-preview.5.23226.2/Sdks/Microsoft.NET.Sdk.BlazorWebAssembly/targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets(160,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/datadisks/disk1/work/B95509FE/w/998E08AE/e/Templates/BaseFolder/AspNet.hhjg2t3oatk0/Client/AspNet.hhjg2t3oatk0.Client.csproj]
2023-04-27T14:13:49.8761830Z [xUnit.net 00:02:17.66]       /datadisks/disk1/work/B95509FE/p/dotnet-cli/sdk/8.0.100-preview.5.23226.2/Sdks/Microsoft.NET.Sdk.BlazorWebAssembly/targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets(160,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/datadisks/disk1/work/B95509FE/w/998E08AE/e/Templates/BaseFolder/AspNet.hhjg2t3oatk0/Client/AspNet.hhjg2t3oatk0.Client.csproj]
2023-04-27T14:13:49.8761891Z [xUnit.net 00:02:17.66]       
2023-04-27T14:13:49.8762018Z [xUnit.net 00:02:17.66]       Expected: True
2023-04-27T14:13:49.8762043Z [xUnit.net 00:02:17.66]       Actual:   False
2023-04-27T14:13:49.8762067Z [xUnit.net 00:02:17.66]       Stack Trace:
2023-04-27T14:13:49.8762089Z [xUnit.net 00:02:17.66]         /_/src/ProjectTemplates/Shared/Project.cs(152,0): at Templates.Test.Helpers.Project.RunDotNetPublishAsync(IDictionary`2 packageOptions, String additionalArgs, Boolean noRestore)
2023-04-27T14:13:49.8762112Z [xUnit.net 00:02:17.66]         /_/src/ProjectTemplates/Shared/BlazorTemplateTest.cs(80,0): at Templates.Blazor.Test.BlazorTemplateTest.CreateBuildPublishAsync(String auth, String[] args, String targetFramework, Boolean serverProject, Boolean onlyCreate)
```

Should we disable these tests (like what was done in https://github.com/dotnet/aspnetcore/pull/47920) to unblock the SDK update?

**Comment by lewing** [comment]
@maraf please take a look

**Comment by maraf** [comment]
Does the publish run in parallel? I see 6 failures with the same path within 2s.
Are the binlogs from tests published? I can't find them.

**Comment by maraf** [comment]
> @eerhardt since this appears to be 6.0 specific I think it is fine to quarantine it while we work on it

It's >= 6.0, the targets are shared.

**Comment by amcasey** [comment]
> > @eerhardt since this appears to be 6.0 specific I think it is fine to quarantine it while we work on it
> 
> It's >= 6.0, the targets are shared.

I'm picking up from @eerhardt on the aspnetcore side and I'm trying to build up some context.  I think @maraf just answered my question but @lewing how did we conclude this was 6.0-specific?

**Comment by amcasey** [comment]
> 



> Does the publish run in parallel? I see 6 failures with the same path within 2s. Are the binlogs from tests published? I can't find them.

I think the template tests _are_ run in parallel.  At least, they muck with the port numbers to avoid conflicts, which you wouldn't expect to be necessary otherwise.

**Comment by wtgodbe** [comment]
> Are the binlogs from tests published? I can't find them.

Unfortunately we don't get access to these, the template tests run from a temp dir and clean themselves up after execution

**Comment by amcasey** [comment]
> Should we disable these tests (like what was done in https://github.com/dotnet/aspnetcore/pull/47920) to unblock the SDK update?

I can't tell from this thread that we think the failures are understood/acceptable enough that we'd want to sweep them under the rug by skipping several tests.  Am I missing something?  Is something important blocked on this PR or is this just a priority because of how far behind it is?

**Comment by lewing** [comment]
> > > @eerhardt since this appears to be 6.0 specific I think it is fine to quarantine it while we work on it
> > 
> > 
> > It's >= 6.0, the targets are shared.
> 
> I'm picking up from @eerhardt on the aspnetcore side and I'm trying to build up some context. I think @maraf just answered my question but @lewing how did we conclude this was 6.0-specific?

There is a reason I asked other people to take a look, I just did a quick pass and clearly misread the targets.  As far as disabling vs blocking do what you think is best, the right people are aware of the issue and looking into it.

**Comment by maraf** [comment]
> I think the template tests are run in parallel. At least, they muck with the port numbers to avoid conflicts, which you wouldn't expect to be necessary otherwise.

This particular obj path is shown in 6 errors within 2s `/datadisks/disk1/work/B95509FE/w/998E08AE/e/Templates/BaseFolder/AspNet.w0wr52oqxukq/Client/obj/Release/net8.0/blazor.publish.boot-extension.json`. Does it mean the single test was run 6 times in parallel?

**Comment by radical** [comment]
`/datadisks/disk1/work/B95509FE/w/998E08AE/e/Templates/BaseFolder/AspNet.w0wr52oqxukq/Client/obj/Release/net8.0/blazor.publish.boot-extension.json`

- What's the root of the test project here? Is the project called `AspNet.*`? And is that suffix generated per test run? If so, then that should isolate the various runs.

**Comment by wtgodbe** [comment]
@MackinnonBuck may be able to help with the template test failures

**Comment by maraf** [comment]
> https://github.com/dotnet/sdk/blob/880798d13942d41e61201e4281604560eade31e9/src/BlazorWasmSdk/Tasks/GenerateBlazorBootExtensionJson.cs#L25-L27 it doesn't seem to be the case. Unless I missed something.

The target that triggers it is referenced in two places
- https://github.com/dotnet/runtime/blob/0f06edefe1588d27de716af5805a20ca918c5f46/src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets#L100
- https://github.com/dotnet/runtime/blob/0f06edefe1588d27de716af5805a20ca918c5f46/src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets#L121

The first one has a guard for nested publish, but the second one added recently with compression doesn't have it.

**Comment by radical** [comment]
This file is also opened by `GenerateWasmBootJson` but is not disposed off:

https://github.com/dotnet/runtime/blob/0f06edefe1588d27de716af5805a20ca918c5f46/src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/GenerateWasmBootJson.cs#L293-L295

`var config = (Dictionary<string, object>)configSerializer.ReadObject(File.OpenRead(configExtension.ItemSpec));`

**Comment by maraf** [comment]
Good catch!

**Comment by radical** [comment]
Preparing a fix in https://github.com/dotnet/runtime/pull/85480 .

**Comment by eerhardt** [comment]
Since the fix is needed in runtime, we will have to wait for that to flow all the way to installer, and update the SDK (and take any _new_ changes that came in the meantime). Would it make sense to disable the tests for now, until the next SDK update?

**Comment by radical** [comment]
A related part of this would be to not run the target twice, which would be a good thing in general. And that should fix the failure here too, if our hypothesis is indeed correct.

**Comment by radical** [comment]
> A related part of this would be to not run the target twice, which would be a good thing in general. And that should fix the failure here too, if our hypothesis is indeed correct.

Ignore me! That file is in `dotnet/runtime` now, so it would need the same flow!
I think we can disable the tests for now.

**Comment by wtgodbe** [comment]
>  Would it make sense to disable the tests for now, until the next SDK update?

I'm supportive, but we should file an issue & link it in the build-ops handoff issue

**Comment by lewing** [comment]
Is there someone that can reproduce this locally to manually verify the fix?

**Pull request merged**
